src/cljs/rems/spa.cljs

```diff
@@ -329,7 +328,7 @@
   [user-and-roles]
   (let [user-and-roles (js->clj user-and-roles :keywordize-keys true)]
     (rf/dispatch-sync [:set-identity (if (:user user-and-roles)
-                                       (update user-and-roles :roles #(mapv keyword (:roles user-and-roles)))
+                                       (update user-and-roles :roles (comp set #(mapv keyword (:roles user-and-roles))))
```

Comment:
this is really weird code, why not just
```
(assoc user-and-roles :roles (set (map keyword (:roles user-and-roles))))

Reply:
I don't know, it's somebody's old code but can be simplified.